R/MeasureFairness.R

```diff
@@ -0,0 +1,54 @@
+#' @title Fairness Measure
+#' @name mlr_measures_fairness
+#'
+#' @description
+#' This measure specializes [Measure] for Fairness Measure problems:
+#' Users could use [groupwise_abs_diff()], [groupwise_diff()], or [groupwise_quotient()] to evaluate
+#' the fairness measures based on base_measures. For example, the false positive rate bias or
+#' the equalized odds ratios. More examples could viewed on mlr3fairness wikipage.
+#'
+#' Predefined measures can be found in the [dictionary][mlr3misc::Dictionary] [mlr_measures].
+#' Predefined operations can be found in the [mlr3fairness].
+#'
+#' @export
+MeasureFairness = R6Class("MeasureFairness", inherit = Measure, cloneable = FALSE,
+  public = list(
+    #' @template field_fun
+    fun = NULL,
+
+    #' @template field_base_measure
+    base_measure = NULL,
+
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    #' @param base_measure The measure used to perform fairness operations.
+    #' @param operation The operation name performed on the base measures.
+    #' @param ps The parameter sets.
+    initialize = function(operation, base_measure, ps = ps()) {
+      super$initialize(
+        id = paste0("fairness.", base_measure$id),
+        range = c(-Inf, Inf),
+        minimize = FALSE,
+        predict_type = base_measure$predict_type,
+        packages = "mlr3fairness",
+        man = paste0("mlr_measures_fairness")
+      )
+      self$fun = get(operation, envir = asNamespace("mlr3fairness"), mode = "function")
+      self$base_measure = base_measure
+    }
+  ),
+
+  private = list(
+    .score = function(prediction, task, ...) {
+      assert_prediction(prediction)
+      if ("requires_task" %in% self$properties && is.null(task)) {
+        stopf("Measure '%s' requires a task", self$id)
+      }
+
```

Comment:
we should enforce some assertion here: 

* task has a single designated column with role `pta`
* there are not more than 2 levels in the `pta` column (AFAICT this measure will not work for more groups, right?)

Reply:
Let us not do the 2nd assertion where we compute the difference. 
Then a `max_abs_diff` measure could e.g. work for >2 groups.

Reply:
Are you arguing to add `max_abs_diff` as operation or to change all operations to operate on more than 2 groups? We can move the assertion to a different place, but it should be performed at some point for operations which do not support multiple groups.

Reply:
I think we could put the assertion into the operation functions. For example, groupwise_abs_diff(). However, should we change the function name to reflect its supported pta? Like binary_abs_diff etc.

Reply:
I have added an assertions.R file to R holder. Which contains assert_binary_pta and assert_pta. assert_binary_pta will call assert_pta everytime and all the fairness operation will call assert_binary_pta to check the binary pta field.